File: php/NewUserEvent.php
<?php

require_once 'IEvent.php';

class NewUserEvent implements IEvent {
    private $eventId;
    protected $userEmail;
    public $userName;

    function __construct(string $userEmail, string $userName) {
        $this->eventId = rand(0, PHP_INT_MAX);
        $this->userEmail = $userEmail;
        $this->userName = $userName;
    }
}

?>

File: java/src/main/java/com/github/yuvallb/sendEventDemo/NewUserEvent.java
package com.github.yuvallb.sendEventDemo;

import java.util.concurrent.ThreadLocalRandom;

public class NewUserEvent implements IEvent {
	
	private int eventId;
	protected String userEmail;
	public String userName;
	
	public NewUserEvent() {
		eventId = ThreadLocalRandom.current().nextInt();
	}
	public NewUserEvent(String userEmail, String userName) {
		this();
		this.userEmail = userEmail;
		this.userName = userName;
	}

	public String getUserEmail() {
		return userEmail;
	}

	public void setUserEmail(String userEmail) {
		this.userEmail = userEmail;
	}

	public String getUserName() {
		return userName;
	}

	public void setUserName(String userName) {
		this.userName = userName;
	}

	public int getEventId() {
		return eventId;
	}
}
